### ISSUE TITLE
[Bug]: Garbage collection performed despite threshold set to "Disabled"

### ISSUE THREAD

**Issue by PineCone227** [Antistasi-Ultimate-Community/A3-Antistasi-Ultimate#145]
### Describe the bug

Garbage collection threshold is disabled in save settings before starting game. Despite this, after large battles the majority of disabled equipment and bodies will despawn before they can be collected (or even before the battle itself ends). Example: Had AAF(Aegis) and AFRF(Aegis) agression on Extreme, massive attacks happened on opposite sides of the map resulting in tens of vehicles (tanks, IFV's, helicopters) being downed or destroyed and hundreds of AI killed. By the time the attacks were over, all vehicles we had managed to disable without complete destruction with the intent of later recovery, were gone. Players also reported bodies they were looting disappearing from in front of them together with the weapons and gear they were trying to recover. 

Expected behaviour: Garbage collection set to "Disabled" should leave all bodies and vehicles laying around until the commander/admin manually presses the cleanup button. 

### How to reproduce

-Play as normal, kick off some agression-raising act to provoke a large response
-Attack/Defend military installation and fight off large numbers of enemies
-Dead enemies and empty vehicles despawn when the clutter piles up, becoming unrecoverable.

### Version

10.5

### Have you altered the code?

No

### Map

Altis

### What server?

Private dedicated server with Headless client(s)

### Mods

```html
<?xml version="1.0" encoding="utf-8"?>
<html>
  <!--Created by Arma 3 Launcher: https://arma3.com-->
  <head>
    <meta name="arma:Type" content="list" />
    <meta name="generator" content="Arma 3 Launcher - https://arma3.com" />
    <title>Arma 3</title>
    <link href="https://fonts.googleapis.com/css?family=Roboto" rel="stylesheet" type="text/css" />
    <style>
body {
	margin: 0;
	padding: 0;
	color: #fff;
	background: #000;	
}

body, th, td {
	font: 95%/1.3 Roboto, Segoe UI, Tahoma, Arial, Helvetica, sans-serif;
}

td {
    padding: 3px 30px 3px 0;
}

h1 {
    padding: 20px 20px 0 20px;
    color: white;
    font-weight: 200;
    font-family: segoe ui;
    font-size: 3em;
    margin: 0;
}

em {
    font-variant: italic;
    color:silver;
}

.before-list {
    padding: 5px 20px 10px 20px;
}

.mod-list {
    background: #222222;
    padding: 20px;
}

.dlc-list {
    background: #222222;
    padding: 20px;
}

.footer {
    padding: 20px;
    color:gray;
}

.whups {
    color:gray;
}

a {
    color: #D18F21;
    text-decoration: underline;
}

a:hover {
    color:#F1AF41;
    text-decoration: none;
}

.from-steam {
    color: #449EBD;
}
.from-local {
    color: gray;
}

</style>
  </head>
  <body>
    <h1>Arma 3 Mods</h1>
    <p class="before-list">
      <em>To import this preset, drag this file onto the Launcher window. Or click the MODS tab, then PRESET in the top right, then IMPORT at the bottom, and finally select this file.</em>
    </p>
    <div class="mod-list">
      <table>
        <tr data-type="ModContainer">
          <td data-type="DisplayName">ace</td>
          <td>
            <span class="from-steam">Steam</span>
          </td>
          <td>
            <a href="http://steamcommunity.com/sharedfiles/filedetails/?id=463939057" data-type="Link">http://steamcommunity.com/sharedfiles/filedetails/?id=463939057</a>
          </td>
        </tr>
        <tr data-type="ModContainer">
          <td data-type="DisplayName">ACE Vehicle Medical</td>
          <td>
            <span class="from-steam">Steam</span>
          </td>
          <td>
            <a href="http://steamcommunity.com/sharedfiles/filedetails/?id=1911374016" data-type="Link">http://steamcommunity.com/sharedfiles/filedetails/?id=1911374016</a>
          </td>
        </tr>
        <tr data-type="ModContainer">
          <td data-type="DisplayName">Animate</td>
          <td>
            <span class="from-steam">Steam</span>
          </td>
          <td>
            <a href="http://steamcommunity.com/sharedfiles/filedetails/?id=2819732932" data-type="Link">http://steamcommunity.com/sharedfiles/filedetails/?id=2819732932</a>
          </td>
        </tr>
        <tr data-type="ModContainer">
          <td data-type="DisplayName">Antistasi Ultimate - Mod</td>
          <td>
            <span class="from-steam">Steam</span>
          </td>
          <td>
            <a href="https://steamcommunity.com/sharedfiles/filedetails/?id=3020755032" data-type="Link">https://steamcommunity.com/sharedfiles/filedetails/?id=3020755032</a>
          </td>
        </tr>
        <tr data-type="ModContainer">
          <td data-type="DisplayName">Arma 3 Aegis</td>
          <td>
            <span class="from-steam">Steam</span>
          </td>
          <td>
            <a href="http://steamcommunity.com/sharedfiles/filedetails/?id=949252631" data-type="Link">http://steamcommunity.com/sharedfiles/filedetails/?id=949252631</a>
          </td>
        </tr>
        <tr data-type="ModContainer">
          <td data-type="DisplayName">Arma 3 Aegis - Marines</td>
          <td>
            <span class="from-steam">Steam</span>
          </td>
          <td>
            <a href="http://steamcommunity.com/sharedfiles/filedetails/?id=2225864448" data-type="Link">http://steamcommunity.com/sharedfiles/filedetails/?id=2225864448</a>
          </td>
        </tr>
        <tr data-type="ModContainer">
          <td data-type="DisplayName">Arma 3 Aegis - Police</td>
          <td>
            <span class="from-steam">Steam</span>
          </td>
          <td>
            <a href="http://steamcommunity.com/sharedfiles/filedetails/?id=2225865544" data-type="Link">http://steamcommunity.com/sharedfiles/filedetails/?id=2225865544</a>
          </td>
        </tr>
        <tr data-type="ModContainer">
          <td data-type="DisplayName">Arma 3 Atlas</td>
          <td>
            <span class="from-steam">Steam</span>
          </td>
          <td>
            <a href="http://steamcommunity.com/sharedfiles/filedetails/?id=2225873516" data-type="Link">http://steamcommunity.com/sharedfiles/filedetails/?id=2225873516</a>
          </td>
        </tr>
        <tr data-type="ModContainer">
          <td data-type="DisplayName">Arma 3 Atlas - Opposing Forces</td>
          <td>
            <span class="from-steam">Steam</span>
          </td>
          <td>
            <a href="https://steamcommunity.com/sharedfiles/filedetails/?id=1873244913" data-type="Link">https://steamcommunity.com/sharedfiles/filedetails/?id=1873244913</a>
          </td>
        </tr>
        <tr data-type="ModContainer">
          <td data-type="DisplayName">Arsenal Search</td>
          <td>
            <span class="from-steam">Steam</span>
          </td>
          <td>
            <a href="http://steamcommunity.com/sharedfiles/filedetails/?id=2060770170" data-type="Link">http://steamcommunity.com/sharedfiles/filedetails/?id=2060770170</a>
          </td>
        </tr>
        <tr data-type="ModContainer">
          <td data-type="DisplayName">CBA_A3</td>
          <td>
            <span class="from-steam">Steam</span>
          </td>
          <td>
            <a href="http://steamcommunity.com/sharedfiles/filedetails/?id=450814997" data-type="Link">http://steamcommunity.com/sharedfiles/filedetails/?id=450814997</a>
          </td>
        </tr>
        <tr data-type="ModContainer">
          <td data-type="DisplayName">CUP Weapons</td>
          <td>
            <span class="from-steam">Steam</span>
          </td>
          <td>
            <a href="http://steamcommunity.com/sharedfiles/filedetails/?id=497660133" data-type="Link">http://steamcommunity.com/sharedfiles/filedetails/?id=497660133</a>
          </td>
        </tr>
        <tr data-type="ModContainer">
          <td data-type="DisplayName">Death and Hit reactions</td>
          <td>
            <span class="from-steam">Steam</span>
          </td>
          <td>
            <a href="https://steamcommunity.com/sharedfiles/filedetails/?id=2993442344" data-type="Link">https://steamcommunity.com/sharedfiles/filedetails/?id=2993442344</a>
          </td>
        </tr>
        <tr data-type="ModContainer">
          <td data-type="DisplayName">Drongo's Better AI Gunners</td>
          <td>
            <span class="from-steam">Steam</span>
          </td>
          <td>
            <a href="https://steamcommunity.com/sharedfiles/filedetails/?id=2917699577" data-type="Link">https://steamcommunity.com/sharedfiles/filedetails/?id=2917699577</a>
          </td>
        </tr>
        <tr data-type="ModContainer">
          <td data-type="DisplayName">dzn Magnifier2PIP</td>
          <td>
            <span class="from-steam">Steam</span>
          </td>
          <td>
            <a href="https://steamcommunity.com/sharedfiles/filedetails/?id=3134718662" data-type="Link">https://steamcommunity.com/sharedfiles/filedetails/?id=3134718662</a>
          </td>
        </tr>
        <tr data-type="ModContainer">
          <td data-type="DisplayName">Enhanced Movement</td>
          <td>
            <span class="from-steam">Steam</span>
          </td>
          <td>
            <a href="http://steamcommunity.com/sharedfiles/filedetails/?id=333310405" data-type="Link">http://steamcommunity.com/sharedfiles/filedetails/?id=333310405</a>
          </td>
        </tr>
        <tr data-type="ModContainer">
          <td data-type="DisplayName">KJW's Two Primary Weapons</td>
          <td>
            <span class="from-steam">Steam</span>
          </td>
          <td>
            <a href="https://steamcommunity.com/sharedfiles/filedetails/?id=2893363164" data-type="Link">https://steamcommunity.com/sharedfiles/filedetails/?id=2893363164</a>
          </td>
        </tr>
        <tr data-type="ModContainer">
          <td data-type="DisplayName">KJW's Two Secondary Weapons</td>
          <td>
            <span class="from-steam">Steam</span>
          </td>
          <td>
            <a href="https://steamcommunity.com/sharedfiles/filedetails/?id=3018159841" data-type="Link">https://steamcommunity.com/sharedfiles/filedetails/?id=3018159841</a>
          </td>
        </tr>
        <tr data-type="ModContainer">
          <td data-type="DisplayName">Simple Armbands</td>
          <td>
            <span class="from-steam">Steam</span>
          </td>
          <td>
            <a href="https://steamcommunity.com/sharedfiles/filedetails/?id=2778578325" data-type="Link">https://steamcommunity.com/sharedfiles/filedetails/?id=2778578325</a>
          </td>
        </tr>
        <tr data-type="ModContainer">
          <td data-type="DisplayName">WMO - Walkable Moving Objects</td>
          <td>
            <span class="from-steam">Steam</span>
          </td>
          <td>
            <a href="http://steamcommunity.com/sharedfiles/filedetails/?id=925018569" data-type="Link">http://steamcommunity.com/sharedfiles/filedetails/?id=925018569</a>
          </td>
        </tr>
        <tr data-type="ModContainer">
          <td data-type="DisplayName">Zeus Enhanced</td>
          <td>
            <span class="from-steam">Steam</span>
          </td>
          <td>
            <a href="http://steamcommunity.com/sharedfiles/filedetails/?id=1779063631" data-type="Link">http://steamcommunity.com/sharedfiles/filedetails/?id=1779063631</a>
          </td>
        </tr>
        <tr data-type="ModContainer">
          <td data-type="DisplayName">Zeus Enhanced - ACE3 Compatibility</td>
          <td>
            <span class="from-steam">Steam</span>
          </td>
          <td>
            <a href="http://steamcommunity.com/sharedfiles/filedetails/?id=2018593688" data-type="Link">http://steamcommunity.com/sharedfiles/filedetails/?id=2018593688</a>
          </td>
        </tr>
      </table>
    </div>
    <div class="footer">
      <span>Created by Arma 3 Launcher by Bohemia Interactive.</span>
    </div>
  </body>
</html>

Additionally: RF and WS CDLC
```

### Additional context

[arma3serverprofiling_x64_2024-03-28_20-11-07.txt](https://github.com/SilenceIsFatto/A3-Antistasi-Ultimate/files/14804848/arma3serverprofiling_x64_2024-03-28_20-11-07.txt)

**Comment by Silence-112** [comment]
The GC script force exits if the param is disabled, so it may be the mission file that has GC enabled.
Did you get hints like "Last Garbage Clean was X ago. Keep track of it."?

**Comment by PineCone227** [comment]
> Did you get hints like "Last Garbage Clean was X ago. Keep track of it."?

Nope, no such hint ever appeared.

**Comment by Silence-112** [comment]
That would explain why changing the parameter has no effect then, because the Antistasi GC isn't doing anything at all.

Which map did this occur on?

**Comment by Silence-112** [comment]
Stale issue, re-open if you have any extra info

**Comment by Drift91** [comment]
I can confirm that this occurs on Altis. I strongly suspect it's the vanilla mission GC doing it, as corpses disappear individually, likely based on how long each individual corpse has existed, as the vanilla system does it. It also takes a long time, about maybe an hour, just like the default value for the vanilla system.

One might think an hour is ample time, but for some people and some battles it isn't sufficient, especially with the fact that stuff that died at the start of the battle has had it's timer ticking the entire fight.

I really think it might be worth adding some of [these commands](https://community.bistudio.com/wiki/Description.ext#Corpse_and_Wreck_Management) to the Description.ext

Btw I've noticed that when I set the vanilla GC to disabled in the in-game Eden editor it still functions, however I believe that's due to the Eden Editor UI not explicitly setting the value in the mission.sqm, resulting in it defaulting to enabled in multiplayer.

**Comment by Drift91** [comment]
So, I've added [this commit](106c50b30501cc6681c4b179f1db9b478d92da38) to my personal fork, and it seems to have had no effect. On top of that, running `[str isInRemainsCollector _this] remoteExec ["systemChat", (owner commanderX)];` in "Targeted Exec" with Zeus Enhanced returns false, so I'm starting to lean more towards it being a feature of Antistasi that's doing this.

**Comment by Silence-112** [comment]
I doubt it is the fn_garbageCleaner functions, they give a hint whenever they run.

If you want to test if they are the ones doing it regardless, set Antistasi GC to disabled and log level to debug.

When a "garbage clean" runs, it should log messages such as `"Cleaning garbage..."`, `"Moving dead solders out of vehicles..."`

Apart from that I don't know what else would be doing it, are you running any extra mods?

**Comment by Drift91** [comment]
I am running other mods, I can't fully recall if I tested with the modpack in a custom mission or not. But I'm definitely not blaming the garbage cleaner function you're talking about, as it's set to be run manually on my server, and the bodies and vehicles don't disappear all at once like they would with that function. I'll have to test again sometime this morning in a custom mission.

**Comment by Drift91** [comment]
I'm pretty sure I just found the responsible function, apologies for taking so long to get around to fully debugging it. Looks like it's `A3A_fnc_postmortem`, which I tested by changing the global variable `cleantime` within it to 10 and subsequently saw a unit I killed despawn in 10 seconds flat. Looking at the code it seems to do exactly what we've been experiencing. Perhaps it might be a good idea to make the function use an admin-configurable duration or allow server admins to entirely disable the function; However it does appear the `cleantime` variable is used in a few other scripts, so maybe use a new variable.
